12-spatial-cv.Rmd

```diff
@@ -69,83 +73,64 @@ Spatial CV\index{cross-validation!spatial CV} alleviates this problem and is the
-  ymn = dem$header$yllcorner,
-  ymx = dem$header$yllcorner + dem$header$nrows * dem$header$cellsize
-  )
-```
-
+There are 175 landslide and 175 non-landslide points, as shown by `summary(lsl$lslpts)`.
+The 175 non-landslide points were sampled randomly from the study area, with the restriction that they must fall outside a small buffer around the landslide polygons.
 
 ```{r lsl-map, echo=FALSE, fig.cap="Landslide initiation points (red) and points unaffected by landsliding (blue) in Southern Ecuador.", fig.scap="Landslide initiation points."}
 # library(tmap)
 # data("lsl", package = "spDataLarge")
-# data("ta", package = "spDataLarge")
-# lsl_sf = st_as_sf(lsl, coords = c("x", "y"), crs = 32717)
-# hs = hillShade(ta$slope * pi / 180, terrain(ta$elev, opt = "aspect"))
-# rect = tmaptools::bb_poly(hs)
-# bbx = tmaptools::bb(hs, xlim = c(-0.02, 1), ylim = c(-0.02, 1), relative = TRUE)
+# ta = terra::rast(system.file("raster/ta.tif", package = "spDataLarge"))
+# lsl_sf = sf::st_as_sf(lsl, coords = c("x", "y"), crs = 32717)
+# hs = terra::shade(slope = ta$slope * pi / 180,
+#                   terra::terrain(ta$elev, v = "aspect", unit = "radians"))
+# # so far tmaptools does not support terra objects
+# 
+# rect = tmaptools::bb_poly(raster::raster(hs))
+# bbx = tmaptools::bb(raster::raster(hs), xlim = c(-0.0001, 1), 
+#                     ylim = c(-0.0001, 1), relative = TRUE)
 # 
-# tm_shape(hs, bbox = bbx) +
-# 	tm_grid(col = "black", n.x = 1, n.y = 1, labels.inside.frame = FALSE,
-# 	        labels.rot = c(0, 90)) +
-# 	tm_raster(palette = gray(0:100 / 100), n = 100, legend.show = FALSE) +
-# 	tm_shape(ta$elev) +
-# 	tm_raster(alpha = 0.5, palette = terrain.colors(10), legend.show = FALSE) +
-# 	tm_shape(lsl_sf) + 
-# 	tm_bubbles("lslpts", size = 0.2, palette = "-RdYlBu", title.col = "Landslide: ") +
-# #   tm_shape(sam) +
-# #   tm_bubbles(border.col = "gold", border.lwd = 2, alpha = 0, size = 0.5) +
+# map = tm_shape(hs, bbox = bbx) +
+#   tm_grid(col = "black", n.x = 1, n.y = 1, labels.inside.frame = FALSE,
+#           labels.rot = c(0, 90)) +
+#   tm_raster(palette = gray(0:100 / 100), n = 100, legend.show = FALSE) +
+#   tm_shape(ta$elev) +
+#   tm_raster(alpha = 0.5, palette = terrain.colors(10), legend.show = FALSE) +
+#   tm_shape(lsl_sf) +
+#   tm_bubbles("lslpts", size = 0.2, palette = "-RdYlBu", 
+#              title.col = "Landslide: ") +
 #   qtm(rect, fill = NULL) +
-# 	tm_layout(outer.margins = c(0.04, 0.04, 0.02, 0.02), frame = FALSE) +
+#   tm_layout(outer.margins = c(0.04, 0.04, 0.02, 0.02), frame = FALSE) +
 #   tm_legend(bg.color = "white")
+# tmap::tmap_save(map, filename = "figures/lsl-map-1.png", width = 11, 
+#                 height = 11, units = "cm")
 knitr::include_graphics("figures/lsl-map-1.png")
 ```
 \index{hillshade}
 
+The first three rows of `lsl`, rounded to two significant digits, can be found in Table \@ref(tab:lslsummary).
+
+```{r lslsummary, echo=FALSE, warning=FALSE}
+lsl_table = lsl |>
```

Comment:
Nice to see the new pipe in there. First step towards #616

Reply:
😃